packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -171,93 +177,122 @@ export class Rule extends Resource implements IRule {
 
     if (targetProps.targetResource) {
       const targetStack = Stack.of(targetProps.targetResource);
-      const targetAccount = targetStack.account;
-      const targetRegion = targetStack.region;
+
+      const targetAccount = (targetProps.targetResource as IResource).env?.account || targetStack.account;
+      const targetRegion = (targetProps.targetResource as IResource).env?.region || targetStack.region;
 
       const sourceStack = Stack.of(this);
       const sourceAccount = sourceStack.account;
       const sourceRegion = sourceStack.region;
 
-      if (targetRegion !== sourceRegion) {
-        throw new Error('Rule and target must be in the same region');
-      }
-
-      if (targetAccount !== sourceAccount) {
-        // cross-account event - strap in, this works differently than regular events!
+      // if the target is in a different account or region and is defined in this CDK App
+      // we can generate all the needed components:
+      // - forwarding rule in the source stack (target: default event bus of the receiver region)
+      // - eventbus permissions policy (creating an extra stack)
+      // - receiver rule in the target stack (target: the actual target)
+      if (!sameEnvDimension(sourceAccount, targetAccount) || !sameEnvDimension(sourceRegion, targetRegion)) {
+        // cross-account and/or cross-region event - strap in, this works differently than regular events!
         // based on:
-        // https://docs.aws.amazon.com/eventbridge/latest/userguide/eventbridge-cross-account-event-delivery.html
+        // https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-cross-account.html
+
+        // for cross-account or cross-region events, we cannot create new components for an imported resource
+        // because we don't have the target stack
+        const isImportedResource = !sameEnvDimension(targetStack.account, targetAccount) || !sameEnvDimension(targetStack.region, targetRegion); //(targetAccount !== targetStack.account) || (targetRegion !== targetStack.region);
+        if (isImportedResource) {
+          throw new Error('Cannot create a cross-account or cross-region rule with an imported resource');
```

Comment:
I guess I'm confused by this check.

The motivation given is "we don't have the target stack", but why do we actually need the target stack? 

* For cross-account access, we could create a support stack, which in fact is done below, so why do we need the target stack? (We could do this opportunistically, create a support stack if we have an imported resource or add to the target stack if we have an owned resource -- and add a flag to disable it maybe?)
* For cross-region access, it seems nothing special needs to be done? Or am I missing something?